File: chronos/src/Chronos.ProfilerAgent/ConfigurationSettings.cpp
#include "StdAfx.h"
#include "ConfigurationSettings.h"


CConfigurationSettings::CConfigurationSettings(void)
{
}

CConfigurationSettings::~CConfigurationSettings(void)
{
	
}

__bool CConfigurationSettings::Initialize(CBaseStream* stream)
{
	//Name
	Name = *CStringMarshaler::Demarshal(stream);
	//Token
	Token = *CStringMarshaler::Demarshal(stream);
	//InitialState
	stream->Read(&InitialState, TypeSize::_BYTE);
	//EventsMask
	stream->Read(&EventsMask, TypeSize::_INT32);
	//TargetProcessName
	TargetProcessName = *CStringMarshaler::Demarshal(stream);
	//ProcessTargetArguments
	ProcessTargetArguments = *CStringMarshaler::Demarshal(stream);
	//ProfileChildProcess
	stream->Read(&ProfileChildProcess, TypeSize::_BOOL);
	//ExclusionsType
	__byte filterType = 0;
	stream->Read(&filterType, TypeSize::_BYTE);
	IsExclusionType = filterType == 1;
	//UseFastHooks
	stream->Read(&UseFastHooks, TypeSize::_BOOL);
	//ProfileSql
	stream->Read(&ProfileSql, TypeSize::_BOOL);
	//Exclusions
	__uint size = 0;
	stream->Read(&size, TypeSize::_INT32);
	FilterItems = new std::vector<std::wstring>();
	for (__uint i = 0; i < size; i++)
	{
		std::wstring filterItem = *CStringMarshaler::Demarshal(stream);
		FilterItems->push_back(filterItem);
	}
	return true;
}

File: chronos/src/Chronos/Core/ConfigurationSettings.cs
using System.Collections.Generic;
using System;

namespace Chronos.Core
{
	[Serializable]
	public class ConfigurationSettings 
	{
		public ConfigurationSettings()
		{
			Name = string.Empty;
			Token = Guid.NewGuid();
			InitialState = SessionState.Closed;
			Events = ClrEventsMask.MonitorNone;
			TargetProcessName = string.Empty;
			ProcessTargetArguments = string.Empty;
			FilterType = FilterType.Include;
			FilterItems = new List<string>();
			UseFastHooks = false;
		    ProfileChildProcess = false;
		    ProfileSql = false;
		}

		public ConfigurationSettings(string name, Guid token, SessionState initialState, ClrEventsMask events, string targetProcessName,
            string processTargetArguments, FilterType filterType, List<string> filterItems, bool useFastHooks, bool profileChildProcess, bool profileSql)
		{
			Name = name;
			Token = token;
			InitialState = initialState;
			Events = events;
			TargetProcessName = targetProcessName;
			ProcessTargetArguments = processTargetArguments;
			FilterType = filterType;
			FilterItems = filterItems;
            UseFastHooks = useFastHooks;
            ProfileChildProcess = profileChildProcess;
		    ProfileSql = profileSql;
		}

		public string Name { get; set; }

		public Guid Token { get; set; }

		public SessionState InitialState { get; set; }

		public ClrEventsMask Events { get; set; }

		public string TargetProcessName { get; set; }

		public string ProcessTargetArguments { get; set; }

		public FilterType FilterType { get; set; }

		public bool UseFastHooks { get; set; }

        public bool ProfileSql { get; set; }

		public List<string> FilterItems { get; set; }

        public bool ProfileChildProcess { get; set; }

		public void RefreshToken()
		{
			Token = Guid.NewGuid();
		}
	}
}
